containerd-shim-spin-config.md

```diff
@@ -0,0 +1,372 @@
+| SpinAppExecutor            | A custom resource definition (CRD) in Kubernetes that represents a Spin executor (such as the Spin containerd shim)                                                                                                                                                          |
+| SpinKube                   | A Kubernetes operator for managing Spin applications                                                                                                                                                                                                                         |
+| Spin runtime               | A Wasm execution engine that executes Wasm components using Wasmtime                                                                                                                                                                                                         |
+| Spin Application Variables | Variables defined in a Spin application manifest (in the `[variables]` section) with values that can be dynamically updated by an [application variables provider](https://developer.fermyon.com/spin/v2/dynamic-configuration#application-variables-runtime-configuration). |
+| Wasmtime                   | The WebAssembly runtime that Spin uses to execute WebAssembly components                                                                                                                                                                                                     |
+
+## Proposal
+
+As laid out in the [background](#background), this SKIP aims to enable the
+following categories of configuration:
+
+- Spin application execution configuration
+- Executor configuration
+
+The first category pertains to the execution of a single Spin application, while
+the later discusses how to configure the platform. 
+
+This SKIP focuses on how the `containerd-shim-spin` and `runwasi` will consume
+configuration. While this SKIP encourages the use of the SpinKube `SpinApp` and
+`SpinAppExecutor` CRDs to set configuration, it does not propose the structure
+of the changes to the CRDs to add configuration. That should be proposed in a
+future SKIP.
+
+### Spin Application Execution Configuration
+
+Spin application execution can be configured via container environment
+variables, which are exposed to the shim. The following is a subset of
+configuration values that may be supported:
+
+| Key                         | Spin CLI                                                    | Example Value           |
+|-----------------------------|-------------------------------------------------------------|-------------------------|
+| SPIN_HTTP_LISTEN_ADDR       | `spin up --listen`                                          | "0.0.0.0:3000"          |
+| OTEL_EXPORTER_OTLP_ENDPOINT | `OTEL_EXPORTER_OTLP_ENDPOINT=http://123.4.5.6:4318 spin up` | "http://123.4.5.6:4318" |
```

Comment:
@Mossaka does this need to be prefixed to not interact with the shim's endpoint?

Reply:
I'm curious @Mossaka if you had any issues testing the new shim otel stuff with the spin shim. Were there any panics or errors?

Reply:
https://github.com/containerd/runwasi/pull/582#issuecomment-2214306550